18_regexp_lab/03.py

```diff
@@ -1,15 +1,26 @@
-"""
-Write a python program that takes a CSV file
-reads it line by line and prints each line
-with first and second columns reversed.
-
-Sample input:
-    Shana,Sargent,shanasargent@isoswitch.com
-    Witt,Hampton,witthampton@zaphire.com
-    Morgan,Grant,morgangrant@lotron.com
-
-Sample output:
-    Sargent,Shana,shanasargent@isoswitch.com
-    Hampton,Witt,witthampton@zaphire.com
-    Grant,Morgan,morgangrant@lotron.com
-"""
+"""
+Elad Agam - Regex
+03 - Replace first and second columns in CSV
+"""
+
+import re
+import argparse
+import os
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-s', '--source')
+args = vars(parser.parse_args())
+
+workingDir =  os.getcwd()
+
+source = os.path.join(workingDir, args['source'])
+
+
+
+
+with open(source,"r") as fin:
+    for line in fin:
+        print  re.sub(r'^(\w+),(\w+),', r'\2,\1,', line)        
+        #I can't understand why the below did not work!!!
+        #print  re.sub(r'^(\w+),(\w+),', lambda m:m.group(1)+","+m.group(0), line)

```

Comment:
Wow - that's true!!

On 26 July 2016 at 21:23, Gabor Szabo <notifications@github.com> wrote:

> In 18_regexp_lab/03.py
> <https://github.com/ynonp/python-examples-verint-2016-07/pull/196#discussion_r72307904>
> :
>
> > +parser = argparse.ArgumentParser()
> > +parser.add_argument('-s', '--source')
> > +args = vars(parser.parse_args())
> > +
> > +workingDir =  os.getcwd()
> > +
> > +source = os.path.join(workingDir, args['source'])
> > +
> > +
> > +
> > +
> > +with open(source,"r") as fin:
> > +    for line in fin:
> > +        print  re.sub(r'^(\w+),(\w+),', r'\2,\1,', line)
> > +        #I can't understand why the below did not work!!!
> > +        #print  re.sub(r'^(\w+),(\w+),', lambda m:m.group(1)+","+m.group(0), line)
>
> group(0) ? didn't you want to use group 1 and 2 there?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ynonp/python-examples-verint-2016-07/pull/196/files/a486e0a95eff0ead6d78fba9827274eb2e88ee79..054bc0dc0a5f704f11e0d746109904232519bdf4#r72307904>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARMSUBrZzpdq5sIhOZtNynDDtaahlUYcks5qZlCWgaJpZM4JTj2Y>
> .
>